packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -892,7 +879,22 @@ function commitUnmount(
             const {destroy, tag} = effect;
             if (destroy !== undefined) {
               if ((tag & HookPassive) !== NoHookEffect) {
-                enqueuePendingPassiveHookEffectUnmount(current, effect);
+                effect.tag |= HookHasEffect;
```

Comment:
I wonder if we can get rid of the rest of this block. I can't remember all the details for how HookHasEffect is used.

(This can be pushed to a follow up. Maybe drop a TODO with context.)

Reply:
What do you mean by, "the rest of this block"?

We still need `PassiveUnmountPendingDev` in DEV in order to avoid over-warning for state updates triggered by passive effect subscriptions.

I'm not sure what you want the TODO to say.

Reply:
> What do you mean by, "the rest of this block"?

Everything inside `if ((tag & HookPassive) !== NoHookEffect) { ... }`

The reason I'm asking is because eventually we want to stop traversing the whole deleted subtree during the  synchronous commit phase. Instead we'd only traverse the paths that truly need to do synchronous clean-up, like ref detaching, `useLayoutEffect` clean up, and `componentWillUnmount`. (Longer term, we want to replace `useLayoutEffect` with a render phase equivalent, and `componentWillUnmount` is class-only. So it's really only ref detaching that we'll need to do.)

> I'm not sure what you want the TODO to say.

Can just be "TODO: Consider if we can move this block out of the synchronous commit phase"

Reply:
Sure. I can add such a comment so we can follow up and consider the change later 👍